src/FanoutGraphqlAppLambdaCallback.ts

```diff
@@ -0,0 +1,74 @@
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+import { ApolloServer, gql } from "apollo-server-lambda";
+import { APIGatewayProxyEvent } from "aws-lambda";
+import { APIGateway } from "aws-sdk";
+import { compose, identity } from "fp-ts/lib/function";
+import ApolloLambdaContextFromPulumiContext from "./ApolloLambdaContextFromPulumiContext";
+import FanoutGraphqlApolloConfig from "./FanoutGraphqlApolloConfig";
+
+type APIGatewayEventMiddleware = (
+  event: APIGatewayProxyEvent,
+) => APIGatewayProxyEvent;
+
+// Will serve graphiql playground. But it has a bug when served at /stage/ on lambda.
+// So in that case, we'll need to patch the event.
+// via: https://github.com/apollographql/apollo-server/pull/2241#issuecomment-460889307
+const playgroundLambdaStageMiddleware: APIGatewayEventMiddleware = (
+  event: APIGatewayProxyEvent,
+): APIGatewayProxyEvent => {
+  const isGetGraphiqlPlayground = event.httpMethod === "GET";
+  if (isGetGraphiqlPlayground) {
+    return { ...event, path: event.requestContext.path || event.path };
+  }
+  // Don't modify event
+  return event;
+};
+
+/**
+ * APIGatewayEventMiddleware that will rewrite events that have base64encoded bodies to not have them.
+ * This is useful because apollo-server-lambda doesn't expect to get base64 encoded bodies. It just tries to JSON.parse(event.body).
+ * Which throws an error and breaks things.
+ */
+const base64DecodeBodyMiddleware: APIGatewayEventMiddleware = event => {
```

Comment:
This is needed to work around this issue I filed against apollo-server-lambda
https://github.com/apollographql/apollo-server/issues/2599